onnxruntime/core/session/custom_ops.cc

```diff
@@ -121,304 +131,392 @@ struct OrtShapeInferContext {
 };
 #endif
 
-ORT_API_STATUS_IMPL(OrtApis::ShapeInferContext_GetInputCount, _In_ const OrtShapeInferContext* context, _Out_ size_t* out) {
+#if ENABLE_ORT_KERNEL_CONTEXT_API
+template <typename T>
+static OrtStatusPtr ExecuteIfKernelContextApiEnabled(const T& fn) {
```

Comment:
I tried using `const std::function<OrtStatusPtr(void)>& fn` initially but that resulted in binary size growth. Using a template seems to allow the compiler/linker to successfully throw away the unused implementation. This occurred with both Windows and Android builds.

Reply:
Not sure. Will leave for now but consider removing if I touch this file again.